test/Airgrab.test.ts

```diff
@@ -0,0 +1,351 @@
+        await expect(
+          airgrab
+            .connect(userSigner)
+            .claim(
+              claimAmount,
+              address,
+              process.env.MERKLE_PROOF!.split(','),
+              proof,
+              validUntil,
+              approvedAt,
+              fractalId,
+            ),
+        ).to.be.revertedWith('Airgrab: already claimed');
+    });
+
+    it('Should delegate voting power when claimed with a delegate', async function () {
+      const userSigner = await ethers.getImpersonatedSigner(testUser);
+      const alice = ((await ethers.getSigners()) as HardhatEthersSigner[])[0]!
+        .address;
+
+      expect(await locking.balanceOf(alice)).to.eq(0);
+      expect(await locking.getVotes(alice)).to.eq(0);
+      expect(await locking.balanceOf(testUser)).to.eq(0);
+      expect(await locking.getVotes(testUser)).to.eq(0);
+      expect(await airgrab.claimed(testUser)).to.eq(false);
+
+      // Claim airgrab with a delegate
+      await airgrab
+        .connect(userSigner)
+        .claim(
+          claimAmount,
+          alice,
+          process.env.MERKLE_PROOF!.split(','),
+          proof,
+          validUntil,
+          approvedAt,
+          fractalId,
+        );
+
+      expect(await locking.balanceOf(alice)).to.eq(claimAmount);
+      expect(await locking.getVotes(alice)).to.eq(claimAmount);
+      expect(await locking.balanceOf(testUser)).to.eq(0);
+      expect(await locking.getVotes(testUser)).to.eq(0);
+      expect(await airgrab.claimed(testUser)).to.eq(true);
+
+      // Claiming again fails
+      await expect(
+        airgrab
+          .connect(userSigner)
+          .claim(
+            claimAmount,
+            address,
+            process.env.MERKLE_PROOF!.split(','),
+            proof,
+            validUntil,
+            approvedAt,
+            fractalId,
+          ),
+      ).to.be.revertedWith('Airgrab: already claimed');
+    });
+  });
+
+  describe('Drain', function () {
+    it('Should fail when airgrab active', async function () {
+      const userSigner = await ethers.getImpersonatedSigner(testUser);
+
+      await expect(
+        airgrab.connect(userSigner).drain(governanceAddresses.MentoToken),
+      ).to.be.revertedWith('Airgrab: not finished');
+    });
+
+    it('Should drain tokens when the airgrab expired', async function () {
+      const userSigner = await ethers.getImpersonatedSigner(testUser);
+
+      const YEAR = 31536000n;
+      await helpers.time.increase(YEAR);
+
+      // token needs to be unpaused if paused
+      if (await mentoToken.paused()) {
+        const governance = await ethers.getImpersonatedSigner(
+          governanceAddresses.TimelockController,
+        );
+        await mentoToken.connect(governance).unpause();
+      }
+
+      const remainingBalance = await mentoToken.balanceOf(
+        governanceAddresses.Airgrab,
+      );
+      await expect(
+        airgrab.connect(userSigner).drain(governanceAddresses.MentoToken),
```

Comment:
not like it's a big difference but maybe add a claim before to make it a bit more realistic?